### PR TITLE
[Flax] Bugfixes in `run_mlm_flax.py`

### PR DESCRIPTION
**Pull request by TevenLeScao**
# What does this PR do?

This PR fixes a few bugs I have observed when using `run_mlm_flax.py`:
- As discussed with @mfuntowicz , `jnp.split` is a lot slower than `np.split` on the first iteration, outright hanging in my tests on simplewiki (~20MB). As this operation doesn't need to be traced. we can use `np.split` instead.
- When using a HF `datasets`, the text column was also passed to the model as input, causing a bug. The PR removes the text column in `dataset.map` to avoid this.
- Finally, using `warmup_steps = 0` (as is default) causes the Flax optimizer to output NaNs. We use 1 as a minimum value for the same warmup-less behaviour.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
LGTM, thanks!

File: examples/language-modeling/run_mlm_flax.py [hunk 1]
```diff
@@ -566,8 +567,9 @@ def tokenize_function(examples):
     ).create(model.params)
 
     # Create learning rate scheduler
+    # warmup_steps = 0 causes the Flax optimizer to return NaNs; warmup_steps = 1 is functionally equivalent.
```
Comment (sgugger) [inline examples/language-modeling/run_mlm_flax.py:570]: We could a do a `logger.warning(...)` if `training_args.warmup_steps` happens to be 0.

Comment (TevenLeScao) [reply examples/language-modeling/run_mlm_flax.py:570]: Just added !
